command_test.go

```diff
@@ -232,6 +232,44 @@ func TestExecuteContextC(t *testing.T) {
 	}
 }
 
+// This tests that the context passed to the root command propagates to its children
+// not only on the first execution but also subsequent calls.
+// Calling the same command multiple times is common when testing cobra applications.
+func TestExecuteContextMultipleTimes(t *testing.T) {
+	ctxRun := func(cmd *Command, args []string) {
+		err := cmd.Context().Err()
```

Comment:
Cool trick to cancel the context for this test.

However, I think we should still test that the we are really receiving the correct context, not just that it hasn't been cancelled.  I think it would cover more cases.

Could you check we have the right context, similarly to the test right above this one?

Reply:
Thanks for the review. I updated it to test for equality.